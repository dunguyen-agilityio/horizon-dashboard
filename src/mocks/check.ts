import { TCheck } from '@/types/check';

export const MOCK_CHECKS: TCheck[] = [
  {
    id: '1',
    name: 'Tony Reichert',
    progress: 45,
    quantity: 20,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '2',
    name: 'Zoey Lang',
    progress: 15,
    quantity: 20,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '3',
    name: 'Jane Fisher',
    progress: 55,
    quantity: 28,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '4',
    name: 'William Howard',
    progress: 65,
    quantity: 30,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
];
