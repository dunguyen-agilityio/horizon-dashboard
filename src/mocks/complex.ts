import { Complex, ComplexStatus } from '@/types/complex';

export const MOCK_COMPLEXES: Complex[] = [
  {
    id: '1',
    name: 'Tony Reichert',
    progress: 45,
    status: ComplexStatus.APPROVED,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '2',
    name: 'Zoey Lang',
    progress: 15,
    status: ComplexStatus.DISABLE,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '3',
    name: 'Jane Fisher',
    progress: 55,
    status: ComplexStatus.ERROR,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '4',
    name: 'William Howard',
    progress: 65,
    status: ComplexStatus.APPROVED,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
];
