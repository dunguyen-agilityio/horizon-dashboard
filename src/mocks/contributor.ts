import { Contributor } from '@/models/Contributor';
import { MOCK_USERS } from './user';

export const MOCK_CONTRIBUTORS = [
  {
    ...MOCK_USERS[0],
    rating: 23,
    template: 'Abstract Colors',
    createdAt: new Date('2024-01-13T07:25:04.188Z'),
  },
  {
    ...MOCK_USERS[1],
    rating: 80,
    template: 'Swipe Circles',
    createdAt: new Date('2024-08-13T07:25:04.188Z'),
  },
  {
    ...MOCK_USERS[2],
    template: 'Colorful Heaven',
    createdAt: new Date('2024-08-11T07:25:04.188Z'),
  },
  {
    ...MOCK_USERS[3],
    template: '3D Cubes Art',
    avatar: undefined,
  },
] as Contributor[];
