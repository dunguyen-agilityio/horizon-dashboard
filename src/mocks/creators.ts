export const MOCK_CREATORS = [
  {
    id: '1',
    avatar: '/default-avatar.webp',
    userName: 'maddison_c21',
    artwork: 9821,
    rating: 90,
  },
  {
    id: '2',
    avatar: '/default-avatar.webp',
    userName: 'karl.willO2',
    artwork: 7032,
    rating: 80,
  },
  {
    id: '3',
    avatar: '/default-avatar.webp',
    userName: 'andreea.1z',
    artwork: 5204,
    rating: 60,
  },
];
