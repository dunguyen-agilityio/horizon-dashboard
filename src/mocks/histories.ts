import { NFT_IMAGES } from '@/constants/images';

export const histories = [
  {
    image: NFT_IMAGES.DEFAULT,
    title: 'Colorful Heaven',
    author: 'Mark Benjamin',
    price: '1.33',
    timeAgo: new Date(Date.now() - 2 * 60 * 60 * 1000),
  },
  {
    image: NFT_IMAGES.DEFAULT,
    title: 'Abstract Colors',
    author: 'Esthera Jackson',
    price: '0.91',
    timeAgo: new Date(Date.now() - 6 * 60 * 60 * 1000),
  },
  {
    image: NFT_IMAGES.DEFAULT,
    title: 'ETH AI Brain',
    author: 'Nick Wilson',
    price: '2.82',
    timeAgo: new Date(Date.now() - 5 * 60 * 60 * 1000),
  },
  {
    image: NFT_IMAGES.DEFAULT,
    title: 'Swipe Circles',
    author: 'Peter Will',
    price: '2.32',
    timeAgo: new Date(Date.now() - 1.5 * 24 * 60 * 60 * 1000),
  },
];
