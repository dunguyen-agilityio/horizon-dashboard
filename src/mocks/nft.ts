// Types
import { NFTResponse, type NFTData } from '@/models/NFT';

// Constants
import { NFT_IMAGES } from '@/constants/images';

// Mocks
import { MOCK_IMAGE_RESPONSE, MOCK_USERS, MOCK_USERS_RESPONSE } from './user';
import { StrapiResponse } from '@/types/strapi';

const [MOCK_USER] = MOCK_USERS;

export const MOCK_NFTS: NFTData[] = [
  {
    id: '1',
    name: 'Mesh Gradients',
    author: MOCK_USER,
    price: 1.23,
    image: { url: NFT_IMAGES.DEFAULT, hash: 'mock hash' },
    members: MOCK_USERS,
    createdAt: new Date('2024-08-17T07:25:04.188Z'),
  },
];

export const MOCK_NFTS_RESPONSE: StrapiResponse<NFTResponse>[] = [
  {
    id: '1',
    attributes: {
      name: 'Mesh Gradients',
      author: { data: MOCK_USERS_RESPONSE[0] },
      price: 1.23,
      members: { data: MOCK_USERS_RESPONSE },
      image: { data: MOCK_IMAGE_RESPONSE },
      createdAt: new Date('2024-08-17T07:25:04.188Z'),
    },
  },
];
