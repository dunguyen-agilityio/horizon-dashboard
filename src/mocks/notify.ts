import { Notify } from '@/models/Notify';

// Type
import { INotification } from '@/types/notification';

export const MOCK_NOTIFIES: Notify[] = [
  new Notify({
    id: '1',
    message: 'Your task is scheduled to start in 10 minutes.',
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
    link: '/kanban/1',
    unread: true,
  }),
  new Notify({
    id: '2',
    message: 'Your task is scheduled to start in 10 minutes.',
    createdAt: new Date('2024-08-13T07:25:04.188Z'),
    link: '/kanban/2',
    unread: true,
  }),
  new Notify({
    id: '3',
    message: 'Your task is scheduled to start in 10 minutes.',
    createdAt: new Date('2024-08-14T02:25:07.188Z'),
    link: '/kanban/3',
  }),
  new Notify({
    id: '4',
    message: 'Your task is scheduled to start in 10 minutes.',
    createdAt: new Date('2024-08-08T08:25:04.188Z'),
    link: '/kanban/4',
  }),
  new Notify({
    id: '5',
    message: 'Your task is scheduled to start in 10 minutes.',
    createdAt: new Date('2024-08-08T08:25:04.188Z'),
    link: '/kanban/5',
  }),
  new Notify({
    id: '6',
    message: 'Your task is scheduled to start in 10 minutes.',
    createdAt: new Date('2024-08-08T08:25:04.188Z'),
    link: '/kanban/6',
  }),
  new Notify({
    id: '7',
    message: 'Your task is scheduled to start in 10 minutes.',
    createdAt: new Date('2024-08-08T08:25:04.188Z'),
    link: '/kanban/7',
  }),
  new Notify({
    id: '8',
    message: 'Your task is scheduled to start in 10 minutes.',
    createdAt: new Date('2024-08-08T08:25:04.188Z'),
    link: '/kanban/8',
  }),
];

export const CONTROL_NOTIFICATIONS: INotification[] = [
  {
    id: '1',
    isSelected: true,
    isDisabled: false,
    label: 'Item update notifications',
  },
  {
    id: '2',
    isSelected: false,
    isDisabled: true,
    label: 'Item comment notifications',
  },
  {
    id: '3',
    isSelected: false,
    isDisabled: true,
    label: 'Buyer review notifications',
  },
  {
    id: '4',
    isSelected: false,
    isDisabled: true,
    label: 'Rating reminders notifications',
  },
  {
    id: '5',
    isSelected: false,
    isDisabled: true,
    label: 'Meetups near you notifications',
  },
  {
    id: '6',
    isSelected: false,
    isDisabled: true,
    label: 'Company news notifications',
  },
  {
    id: '7',
    isSelected: false,
    isDisabled: true,
    label: 'New launches and projects',
  },
  {
    id: '8',
    isSelected: false,
    isDisabled: true,
    label: 'Monthly product changes',
  },
  {
    id: '9',
    isSelected: false,
    isDisabled: true,
    label: 'Subscribe to newsletter',
  },
  {
    id: '10',
    isSelected: true,
    isDisabled: true,
    label: 'Email me when someone follows me',
  },
];
