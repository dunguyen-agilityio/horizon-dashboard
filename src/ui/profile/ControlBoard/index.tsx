// Components
import { BoxIcon, Text } from '@/components';
import { Switch } from '@nextui-org/switch';

// Icons
import More from '@/icons/More';

// Types
import { TEXT_SIZE } from '@/types/text';

// Constants
import { CONTROL_NOTIFICATIONS } from '@/mocks/notify';

const ControlBoard = () => (
  <div className="dark:bg-indigo bg-white w-[350px] xl:w-[421px] pt-[23px] p-[31px] rounded-md">
    <div className="flex justify-between">
      <Text size={TEXT_SIZE.extra} className="font-extrabold leading-8 mb-4">
        Notifications
      </Text>
      <BoxIcon
        icon={<More />}
        customClass="rotate-90 w-9 h-9 bg-gray dark:bg-indigo-light rounded-lg fill-blue-450 dark:fill-white"
      />
    </div>
    <div className="flex flex-col gap-[22px] ml-1">
      {CONTROL_NOTIFICATIONS.map(({ id, label, isSelected, isDisabled }) => (
        <Switch
          key={`switch-${id}`}
          defaultSelected={isSelected}
          isDisabled={isDisabled}
          size="sm"
          classNames={{
            label: 'ml-2 text-primary dark:text-white text-sm',
            wrapper:
              'group-data-[selected=true]:bg-blue-450 group-data-[selected=true]:dark:bg-purple-750',
          }}
        >
          {label}
        </Switch>
      ))}
    </div>
  </div>
);

export default ControlBoard;
