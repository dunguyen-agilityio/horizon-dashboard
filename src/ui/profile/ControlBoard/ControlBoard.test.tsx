import { render } from '@testing-library/react';

// Components
import ControlBoard from '.';

// Constants
import { CONTROL_NOTIFICATIONS } from '@/mocks/notify';

describe('ControlBoard component', () => {
  it('ControlBoard should render match snapshot', () => {
    const { container } = render(<ControlBoard />);
    expect(container).toMatchSnapshot();
  });

  it('ControlBoard should render correct title', () => {
    const { getByText } = render(<ControlBoard />);

    const title = getByText('Notifications');
    expect(title).toBeInTheDocument();
  });

  it('ControlBoard should render correct label', () => {
    const { getByText } = render(<ControlBoard />);

    const label = getByText(CONTROL_NOTIFICATIONS[1].label);
    expect(label).toBeInTheDocument();
  });

  it('ControlBoard should render switch button', () => {
    const { getAllByRole } = render(<ControlBoard />);

    const listSwitchButton = getAllByRole('switch');
    expect(listSwitchButton[0]).toBeInTheDocument();
  });
});
