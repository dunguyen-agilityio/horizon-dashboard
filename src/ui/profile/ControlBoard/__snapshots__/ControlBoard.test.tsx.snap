// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlBoard component ControlBoard should render match snapshot 1`] = `
<div>
  <div
    class="dark:bg-indigo bg-white pt-[23px] p-[31px] rounded-md"
  >
    <div
      class="flex justify-between"
    >
      <p
        class="text-primary dark:text-white text-1.5xl leading-xl font-extrabold leading-8 mb-4 2xl:mb-2"
      >
        Notifications
      </p>
      <i
        class="flex justify-center items-center hover:cursor-pointer rotate-90 w-9 h-9 bg-gray dark:bg-indigo-light rounded-lg fill-blue-450 dark:fill-white"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_201_3969)"
          >
            <path
              d="M12 8C13.1 8 14 7.1 14 6C14 4.9 13.1 4 12 4C10.9 4 10 4.9 10 6C10 7.1 10.9 8 12 8ZM12 10C10.9 10 10 10.9 10 12C10 13.1 10.9 14 12 14C13.1 14 14 13.1 14 12C14 10.9 13.1 10 12 10ZM12 16C10.9 16 10 16.9 10 18C10 19.1 10.9 20 12 20C13.1 20 14 19.1 14 18C14 16.9 13.1 16 12 16Z"
            />
          </g>
          <defs>
            <clippath
              id="clip0_201_3969"
            >
              <rect
                fill="white"
                height="24"
                width="24"
              />
            </clippath>
          </defs>
        </svg>
      </i>
    </div>
    <div
      class="flex flex-col gap-6 ml-1"
    >
      <label
        class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer touch-none tap-highlight-transparent"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-label="Item update notifications"
            aria-labelledby=":r0:"
            role="switch"
            type="checkbox"
            value=""
          />
        </span>
        <span
          aria-hidden="true"
          class="px-1 relative inline-flex items-center justify-start flex-shrink-0 overflow-hidden bg-default-200 rounded-full outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:text-primary-foreground w-10 h-6 mr-2 rtl:ml-2 rtl:mr-[unset] transition-background group-data-[selected=true]:bg-blue-450 group-data-[selected=true]:dark:bg-purple-750"
        >
          <span
            class="z-10 flex items-center justify-center bg-white shadow-small rounded-full origin-right w-4 h-4 text-tiny group-data-[selected=true]:ml-4 rtl:group-data-[selected=true]:ml-0 rtl:group-data-[selected=true]:mr-4 transition-all group-data-[pressed=true]:w-5 group-data-[selected]:group-data-[pressed]:ml-3"
          />
        </span>
        <span
          class="relative select-none ml-2 text-primary dark:text-white text-sm"
          id=":r0:"
        >
          Item update notifications
        </span>
      </label>
      <label
        class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer touch-none tap-highlight-transparent opacity-disabled pointer-events-none"
        data-disabled="true"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-label="Item comment notifications"
            aria-labelledby=":r1:"
            disabled=""
            role="switch"
            type="checkbox"
            value=""
          />
        </span>
        <span
          aria-hidden="true"
          class="px-1 relative inline-flex items-center justify-start flex-shrink-0 overflow-hidden bg-default-200 rounded-full outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:text-primary-foreground w-10 h-6 mr-2 rtl:ml-2 rtl:mr-[unset] transition-background group-data-[selected=true]:bg-blue-450 group-data-[selected=true]:dark:bg-purple-750"
        >
          <span
            class="z-10 flex items-center justify-center bg-white shadow-small rounded-full origin-right w-4 h-4 text-tiny group-data-[selected=true]:ml-4 rtl:group-data-[selected=true]:ml-0 rtl:group-data-[selected=true]:mr-4 transition-all group-data-[pressed=true]:w-5 group-data-[selected]:group-data-[pressed]:ml-3"
          />
        </span>
        <span
          class="relative select-none ml-2 text-primary dark:text-white text-sm"
          id=":r1:"
        >
          Item comment notifications
        </span>
      </label>
      <label
        class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer touch-none tap-highlight-transparent opacity-disabled pointer-events-none"
        data-disabled="true"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-label="Buyer review notifications"
            aria-labelledby=":r2:"
            disabled=""
            role="switch"
            type="checkbox"
            value=""
          />
        </span>
        <span
          aria-hidden="true"
          class="px-1 relative inline-flex items-center justify-start flex-shrink-0 overflow-hidden bg-default-200 rounded-full outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:text-primary-foreground w-10 h-6 mr-2 rtl:ml-2 rtl:mr-[unset] transition-background group-data-[selected=true]:bg-blue-450 group-data-[selected=true]:dark:bg-purple-750"
        >
          <span
            class="z-10 flex items-center justify-center bg-white shadow-small rounded-full origin-right w-4 h-4 text-tiny group-data-[selected=true]:ml-4 rtl:group-data-[selected=true]:ml-0 rtl:group-data-[selected=true]:mr-4 transition-all group-data-[pressed=true]:w-5 group-data-[selected]:group-data-[pressed]:ml-3"
          />
        </span>
        <span
          class="relative select-none ml-2 text-primary dark:text-white text-sm"
          id=":r2:"
        >
          Buyer review notifications
        </span>
      </label>
      <label
        class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer touch-none tap-highlight-transparent opacity-disabled pointer-events-none"
        data-disabled="true"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-label="Rating reminders notifications"
            aria-labelledby=":r3:"
            disabled=""
            role="switch"
            type="checkbox"
            value=""
          />
        </span>
        <span
          aria-hidden="true"
          class="px-1 relative inline-flex items-center justify-start flex-shrink-0 overflow-hidden bg-default-200 rounded-full outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:text-primary-foreground w-10 h-6 mr-2 rtl:ml-2 rtl:mr-[unset] transition-background group-data-[selected=true]:bg-blue-450 group-data-[selected=true]:dark:bg-purple-750"
        >
          <span
            class="z-10 flex items-center justify-center bg-white shadow-small rounded-full origin-right w-4 h-4 text-tiny group-data-[selected=true]:ml-4 rtl:group-data-[selected=true]:ml-0 rtl:group-data-[selected=true]:mr-4 transition-all group-data-[pressed=true]:w-5 group-data-[selected]:group-data-[pressed]:ml-3"
          />
        </span>
        <span
          class="relative select-none ml-2 text-primary dark:text-white text-sm"
          id=":r3:"
        >
          Rating reminders notifications
        </span>
      </label>
      <label
        class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer touch-none tap-highlight-transparent opacity-disabled pointer-events-none"
        data-disabled="true"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-label="Meetups near you notifications"
            aria-labelledby=":r4:"
            disabled=""
            role="switch"
            type="checkbox"
            value=""
          />
        </span>
        <span
          aria-hidden="true"
          class="px-1 relative inline-flex items-center justify-start flex-shrink-0 overflow-hidden bg-default-200 rounded-full outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:text-primary-foreground w-10 h-6 mr-2 rtl:ml-2 rtl:mr-[unset] transition-background group-data-[selected=true]:bg-blue-450 group-data-[selected=true]:dark:bg-purple-750"
        >
          <span
            class="z-10 flex items-center justify-center bg-white shadow-small rounded-full origin-right w-4 h-4 text-tiny group-data-[selected=true]:ml-4 rtl:group-data-[selected=true]:ml-0 rtl:group-data-[selected=true]:mr-4 transition-all group-data-[pressed=true]:w-5 group-data-[selected]:group-data-[pressed]:ml-3"
          />
        </span>
        <span
          class="relative select-none ml-2 text-primary dark:text-white text-sm"
          id=":r4:"
        >
          Meetups near you notifications
        </span>
      </label>
      <label
        class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer touch-none tap-highlight-transparent opacity-disabled pointer-events-none"
        data-disabled="true"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-label="Company news notifications"
            aria-labelledby=":r5:"
            disabled=""
            role="switch"
            type="checkbox"
            value=""
          />
        </span>
        <span
          aria-hidden="true"
          class="px-1 relative inline-flex items-center justify-start flex-shrink-0 overflow-hidden bg-default-200 rounded-full outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:text-primary-foreground w-10 h-6 mr-2 rtl:ml-2 rtl:mr-[unset] transition-background group-data-[selected=true]:bg-blue-450 group-data-[selected=true]:dark:bg-purple-750"
        >
          <span
            class="z-10 flex items-center justify-center bg-white shadow-small rounded-full origin-right w-4 h-4 text-tiny group-data-[selected=true]:ml-4 rtl:group-data-[selected=true]:ml-0 rtl:group-data-[selected=true]:mr-4 transition-all group-data-[pressed=true]:w-5 group-data-[selected]:group-data-[pressed]:ml-3"
          />
        </span>
        <span
          class="relative select-none ml-2 text-primary dark:text-white text-sm"
          id=":r5:"
        >
          Company news notifications
        </span>
      </label>
      <label
        class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer touch-none tap-highlight-transparent opacity-disabled pointer-events-none"
        data-disabled="true"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-label="New launches and projects"
            aria-labelledby=":r6:"
            disabled=""
            role="switch"
            type="checkbox"
            value=""
          />
        </span>
        <span
          aria-hidden="true"
          class="px-1 relative inline-flex items-center justify-start flex-shrink-0 overflow-hidden bg-default-200 rounded-full outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:text-primary-foreground w-10 h-6 mr-2 rtl:ml-2 rtl:mr-[unset] transition-background group-data-[selected=true]:bg-blue-450 group-data-[selected=true]:dark:bg-purple-750"
        >
          <span
            class="z-10 flex items-center justify-center bg-white shadow-small rounded-full origin-right w-4 h-4 text-tiny group-data-[selected=true]:ml-4 rtl:group-data-[selected=true]:ml-0 rtl:group-data-[selected=true]:mr-4 transition-all group-data-[pressed=true]:w-5 group-data-[selected]:group-data-[pressed]:ml-3"
          />
        </span>
        <span
          class="relative select-none ml-2 text-primary dark:text-white text-sm"
          id=":r6:"
        >
          New launches and projects
        </span>
      </label>
      <label
        class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer touch-none tap-highlight-transparent opacity-disabled pointer-events-none"
        data-disabled="true"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-label="Monthly product changes"
            aria-labelledby=":r7:"
            disabled=""
            role="switch"
            type="checkbox"
            value=""
          />
        </span>
        <span
          aria-hidden="true"
          class="px-1 relative inline-flex items-center justify-start flex-shrink-0 overflow-hidden bg-default-200 rounded-full outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:text-primary-foreground w-10 h-6 mr-2 rtl:ml-2 rtl:mr-[unset] transition-background group-data-[selected=true]:bg-blue-450 group-data-[selected=true]:dark:bg-purple-750"
        >
          <span
            class="z-10 flex items-center justify-center bg-white shadow-small rounded-full origin-right w-4 h-4 text-tiny group-data-[selected=true]:ml-4 rtl:group-data-[selected=true]:ml-0 rtl:group-data-[selected=true]:mr-4 transition-all group-data-[pressed=true]:w-5 group-data-[selected]:group-data-[pressed]:ml-3"
          />
        </span>
        <span
          class="relative select-none ml-2 text-primary dark:text-white text-sm"
          id=":r7:"
        >
          Monthly product changes
        </span>
      </label>
      <label
        class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer touch-none tap-highlight-transparent opacity-disabled pointer-events-none"
        data-disabled="true"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-label="Subscribe to newsletter"
            aria-labelledby=":r8:"
            disabled=""
            role="switch"
            type="checkbox"
            value=""
          />
        </span>
        <span
          aria-hidden="true"
          class="px-1 relative inline-flex items-center justify-start flex-shrink-0 overflow-hidden bg-default-200 rounded-full outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:text-primary-foreground w-10 h-6 mr-2 rtl:ml-2 rtl:mr-[unset] transition-background group-data-[selected=true]:bg-blue-450 group-data-[selected=true]:dark:bg-purple-750"
        >
          <span
            class="z-10 flex items-center justify-center bg-white shadow-small rounded-full origin-right w-4 h-4 text-tiny group-data-[selected=true]:ml-4 rtl:group-data-[selected=true]:ml-0 rtl:group-data-[selected=true]:mr-4 transition-all group-data-[pressed=true]:w-5 group-data-[selected]:group-data-[pressed]:ml-3"
          />
        </span>
        <span
          class="relative select-none ml-2 text-primary dark:text-white text-sm"
          id=":r8:"
        >
          Subscribe to newsletter
        </span>
      </label>
      <label
        class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer touch-none tap-highlight-transparent opacity-disabled pointer-events-none"
        data-disabled="true"
        data-selected="true"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-label="Email me when someone follows me"
            aria-labelledby=":r9:"
            checked=""
            disabled=""
            role="switch"
            type="checkbox"
            value=""
          />
        </span>
        <span
          aria-hidden="true"
          class="px-1 relative inline-flex items-center justify-start flex-shrink-0 overflow-hidden bg-default-200 rounded-full outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:text-primary-foreground w-10 h-6 mr-2 rtl:ml-2 rtl:mr-[unset] transition-background group-data-[selected=true]:bg-blue-450 group-data-[selected=true]:dark:bg-purple-750"
        >
          <span
            class="z-10 flex items-center justify-center bg-white shadow-small rounded-full origin-right w-4 h-4 text-tiny group-data-[selected=true]:ml-4 rtl:group-data-[selected=true]:ml-0 rtl:group-data-[selected=true]:mr-4 transition-all group-data-[pressed=true]:w-5 group-data-[selected]:group-data-[pressed]:ml-3"
          />
        </span>
        <span
          class="relative select-none ml-2 text-primary dark:text-white text-sm"
          id=":r9:"
        >
          Email me when someone follows me
        </span>
      </label>
    </div>
  </div>
</div>
`;
